test/e2e/specs/editor/various/block-switcher-test.spec.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Block Switcher', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'Block variation transforms', async ( { editor, page } ) => {
+		// This is the `stack` Group variation.
+		await editor.insertBlock( {
+			name: 'core/group',
+			attributes: {
+				layout: {
+					type: 'flex',
+					orientation: 'vertical',
+				},
+			},
+			innerBlocks: [
+				{
+					name: 'core/paragraph',
+					attributes: { content: '1' },
+				},
+			],
+		} );
+		// Transform to `Stack` variation.
+		await editor.clickBlockToolbarButton( 'Stack' );
+		const variations = page
+			.getByRole( 'menu', { name: 'Stack' } )
+			.getByRole( 'group' );
+		await expect(
+			variations.getByRole( 'menuitem', { name: 'Stack' } )
+		).toBeHidden();
+		await variations.getByRole( 'menuitem', { name: 'Row' } ).click();
+		expect( await editor.getBlocks() ).toMatchObject( [
```

Comment:
Nit, more of an FYI: We can use `expect.poll` to auto repetitively check for the result:

```suggestion
		await expect( editor.getBlocks ).toMatchObject( [
```